crates/rpc/rpc/src/eth/api/traits/transaction.rs

```diff
@@ -56,11 +68,35 @@ pub type StateCacheDB = CacheDB<StateProviderDatabase<StateProviderBox>>;
 /// This implementation follows the behaviour of Geth and disables the basefee check for tracing.
 #[async_trait::async_trait]
 pub trait EthTransactions: Send + Sync {
+    /// Transaction pool with pending transactions. [`TransactionPool::Transaction`] is the
+    /// supported transaction type.
+    type Pool: TransactionPool;
+
     /// Returns a handle for reading data from disk.
     ///
     /// Data access in default (L1) trait method implementations.
     fn provider(&self) -> &impl BlockReaderIdExt;
 
+    /// Returns a handle for reading data from memory.
+    ///
+    /// Data access in default (L1) trait method implementations.
+    fn cache(&self) -> &EthStateCache;
+
+    /// Returns a handle for reading data from pool.
+    ///
+    /// Data access in default (L1) trait method implementations.
+    fn pool(&self) -> &Self::Pool;
+
+    /// Returns a handle for forwarding received raw transactions.
+    ///
+    /// Access to transaction forwarder in default (L1) trait method implementations.
+    fn raw_tx_forwarder(&self) -> &Option<Arc<dyn RawTransactionForwarder>>;
+
+    /// Returns a handle for reading evm config.
+    ///
+    /// Data access in default (L1) trait method implementations.
+    fn evm_config(&self) -> &impl ConfigureEvm;
```

Comment:
I'd also like to auto_impl for & if possible

Reply:
it's not possible, but not because `atuo_impl` fails (which is the case for `SpawnBlocking`, which has methods taking a closure with `Self` as arg). I get this error, which I have a hunch is related to this https://github.com/rust-lang/rust/issues/100013, since `ConfigureEvm` trait uses GAT of specific lifetime.

```
error: lifetime may not live long enough
  --> crates/rpc/rpc/src/eth/api/transactions.rs:80:9
   |
79 |     fn evm_config(&self) -> impl ConfigureEvm + '_ {
   |                   - let's call the lifetime of this reference `'1`
80 |         self.inner.evm_config()
   |         ^^^^^^^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'static`
```